disnake/ui/modal.py

```diff
@@ -138,7 +120,62 @@ def add_component(self, component: Union[InputText, List[InputText]]) -> None:
                 raise TypeError(
                     f"component must be of type InputText or a list of InputText, not {c.__class__.__name__}."
                 )
-            self._underlying.components.append(c)
+            try:
+                self.components[-1].append_item(c)
+            except (ValueError, IndexError):
+                self.components.append(ActionRow(c))
+
+    def add_input_text(
+        self,
+        *,
+        label: str,
+        custom_id: str,
+        style: InputTextStyle = InputTextStyle.short,
+        placeholder: Optional[str] = None,
+        value: Optional[str] = None,
+        required: bool = True,
+        min_length: int = 0,
+        max_length: Optional[int] = None,
+    ):
+        """Adds an input text component to the modal.
+
+        To append a pre-existing :class:`disnake.ui.InputText` use the
+        :meth:`append_component` method instead.
+
+        Parameters
+        -----------
+        style: :class:`.InputTextStyle`
+            The style of the input text.
```

Comment:
I know you copy-pasted the docstrings, but the same suggestion I did in `ui.InputText` goes here.